File: dainf-lab-server/src/main/java/br/edu/utfpr/dainf/storage/StorageService.java
package br.edu.utfpr.dainf.storage;

public interface StorageService {

    /**
     * Generates a signed URL for accessing a file.
     */
    String getSignedUrl(String bucketName, String objectName, int expirationInSeconds, String method);

    /**
     * Moves a file from a temporary folder to a permanent folder.
     * Each bucket has a temporary folder inside it.
     */
    void moveToPermanentFolder(String bucket, String oldObjectName, String newObjectName);
}

File: dainf-lab-client/src/app/shared/storage/storage.service.ts
import { Observable } from 'rxjs';
import { Image } from './image';

export interface StorageService {
  getSignedUrl(
    objectName: string,
    method: 'GET' | 'PUT' | 'DELETE',
  ): Observable<string>;

  get(objectName: string): Observable<Blob>;

  upload(file: File): Observable<Image>;

  delete(objectName: string): Observable<void>;
}
